Tests/LogtoClientTests/LogtoClient/LogtoClientTests+SignIn.swift

```diff
@@ -0,0 +1,83 @@
+import Logto
+@testable import LogtoClient
+import LogtoMock
+import XCTest
+
+class LogtoAuthSessionSuccessMock: LogtoAuthSession {
+    override func start() {
+        completion(.success(response: try! JSONDecoder().decode(LogtoCore.CodeTokenResponse.self, from: Data("""
+        {
+            "accessToken": "foo",
+            "refreshToken": "bar",
+            "idToken": "baz",
+            "scope": "openid offline_access",
+            "expiresIn": 300
+        }
+        """.utf8))))
+    }
+}
+
+class LogtoAuthSessionFailureMock: LogtoAuthSession {
+    override func start() {
+        completion(.failure(error: LogtoAuthSession.Errors.SignIn(type: .unknownError, innerError: nil)))
+    }
+}
+
+extension LogtoClientTests {
+    func testSignInOk() throws {
+        let client = buildClient()
+        let expectOk = expectation(description: "Sign in OK")
+
+        client.signInWithBrowser(
+            authSessionType: LogtoAuthSessionSuccessMock.self,
+            redirectUri: "io.logto.dev://callback"
+        ) {
+            XCTAssertNil($0)
+            XCTAssertEqual(client.idToken, "baz")
+            XCTAssertEqual(client.refreshToken, "bar")
+            XCTAssertEqual(client.accessTokenMap[client.buildAccessTokenKey(for: nil, scopes: [])]?.token, "foo")
```

Comment:
nit: not a big deal but mostly a shared const variable or input value reference is preferred instead of asserting a const value directly.  Just in case the input value may be updated mistakenly, or hard to locate.